docs: improve cross-referencing with intersphinx (#203)
ianhi/mpl-interactions: docs/index.rst

To achieve this, mpl_interactions provides:

* A way to control the output of pyplot functions (e.g. :func:`~matplotlib.pyplot.plot` and :func:`~matplotlib.pyplot.hist`) with sliders.
* A function to compare horizontal and vertical slices of heatmaps.
* A function allowing zooming using the scroll wheel.

Installation
^^^^^^^^^^^^^

To install, simply run:

.. code-block::

   pip install mpl_interactions

To also install version of ``ipympl`` and :doc:`ipywidgets <ipywidgets:index>` that are known to work install the optional jupyter dependencies by running:

.. code-block::

   pip install mpl_interactions[jupyter]

Further instructions for installation from JupyterLab can be found on the :doc:`Installation` page. 

Getting Help
^^^^^^^^^^^^

> If you have a question on how to do something with `mpl_interactions` a great place

Reviewer: ```suggestion
If you have a question on using this library a great place
```

Reviewer: Avoids some weird linking

Author: Hehe, that linking is the point of this PR: all backticked text in rst files become links to an API, either that of mpl_interactions or an external one from intersphinx. So the link in this is to the API page for this module. But I agree that a link to the API is a bit weird here.
